### PR TITLE
Add support for distributing host keys

### LINKED ISSUE #4: SSH public keys are not valid ssh host public keys
I had a minor issue with this, which is that by default the known_hosts public key values are being set to the passed-in SSH public key: https://github.com/oasis-roles/passwordless_ssh/blob/master/tasks/main.yml#L37. As the topic would indicate, SSH public keys and SSH *host* public keys are not interchangeable. The host key must be the public component of one of the host private keys ideally known only to a given SSH host.

The SSH *host* keys live in `/etc/ssh`, e.g. `/etc/ssh/ssh_host_rsa_key` and `/etc/ssh/ssh_host_rsa_key.pub`; each known_hosts entry's key should be the contents of one of these public host keys on the corresponding host for that known_hosts entry.

### PR DESCRIPTION
**Pull request by tehsmyers**
This is arguably a backward-incompatible change since it no longer
distributes host keys by default, but as the referenced issue would
indicate, this role was not actually distributing host keys. Therefore,
I consider this PR to be an enhancement.

fixes #4

### CODE REVIEW

File: tasks/main.yml
Comment (greg-hellings) [inline tasks/main.yml:33]: I'm not so sure that this block should *replace* the above block, but rather should add to it. If I try to ssh as user `ghelling` from my localhost to a remote session, I need my user's key to be added to the `.ssh/known_hosts` file of the remote system, and not my laptop's /etc/ssh/ key. That allows me to login as my user to that remote user without password.

Does this change allow any user on the current system to login to the remote system without a password? I've never before touched nor needed to touch anything under `/etc/ssh` so I'm not sure what removing my change and adding yours will accomplish.

**Review by greg-hellings — CHANGES_REQUESTED** [review]
Requesting comment, not necessary change.

**Comment by tehsmyers** [comment]
I ran the openstack tests manually with a good result; all of jenkins is showing red right now due (I assume) to the ongoing maintenance of some of its supporting services.

File: tasks/main.yml
Comment (tehsmyers) [reply tasks/main.yml:33]: This change does nothing with the `authorized_keys` handling before it, and does nothing to disallow the ability for a user to log in to a remote system without a password.

`known_hosts` has nothing to do with users, login, or the host you're connecting from. `known_hosts` is a list of host keys (never user pubkeys) that SSH uses to authenticate *hosts* that you're connecting to before user authentication begins. You do not need your user's key to be added to `known_hosts` at any time.

**Review by greg-hellings — APPROVED** [review]

**Comment by greg-hellings** [comment]
This role is also used by the `gluster_gdeploy` tests, where it was important to get the keys properly configured. As long as this change doesn't break that role, then we should be good to go.

**Comment by tehsmyers** [comment]
Yeah, that should still be fine. The previous host key handling would have had no effect, and playbooks with the same vars as before will continue to have no effect. If failing host key verification starts causing problems, you can either set the host_keys using `passwordless_ssh_host_keys: [...]`, or autogather them with `passwordless_ssh_gather_host_keys: true`.

**Comment by tehsmyers** [comment]
(That said, I'm still going to run the `gluster_gdeploy` tests to be sure before hitting the big merge button)

**Review by robled — APPROVED** [review]

**Comment by tehsmyers** [comment]
gluster_deploy appears to be unaffacted by this change

**Pull request merged**
